File: leetcodeContest/26周双周赛/3.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function (root) {
  let res = 0
  const dfs = (root, path) => {
    if (!root) {
      return
    }
    let max = path.length === 0 ? -999999 : Math.max(...path)
    if (root.val > max) {
      res ++;
    }
    path.push(root.val)
    if (root.left) {
      dfs(root.left, path)
    }
    if (root.right) {
      dfs(root.right, path)
    }
    path.pop()
  }
  dfs(root, [])
  return res;
}

File: leetcodeContest/190周/3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pseudoPalindromicPaths (self, root: TreeNode) -> int:
        self.a = [0] * 10
        self.count = 0
        self.dfs(root, [])
        return self.count;
        
    def dfs(self, root, path):
        if root is None:
            return
        self.a[root.val]+=1
        if root.left is None and root.right is None:
            md = 0
            for i in range(10):
                if self.a[i] % 2 != 0 and self.a[i] != 0:
                    md +=1
            print(self.a)        
            if md < 2:
                self.count +=1
        if root.left:
            self.dfs(root.left, path)
        if root.right:
            self.dfs(root.right, path)
        self.a[root.val] -=1
        